algosdk/future/transaction.py

```diff
@@ -477,6 +483,161 @@ def __eq__(self, other):
+            "votefst": self.votefst,
+            "votekd": self.votekd,
+            "votekey": base64.b64decode(self.votepk) if self.votepk is not None else None,
+            "votelst": self.votelst,
+        }
+        d.update(super(OnlineKeyregTxn, self).dictify())
+        od = OrderedDict(sorted(d.items()))
+
+        return od
+
+    @staticmethod
+    def _undictify(d):
+        votekey = None
+        selkey = None
+        votefst = None
+        votelst = None
+        votekd = None
+
+        if "votekey" in d:
+            votekey = base64.b64encode(d["votekey"]).decode()
+        if "selkey" in d:
+            selkey = base64.b64encode(d["selkey"]).decode()
+        if "votefst" in d:
+            votefst = d["votefst"]
+        if "votelst" in d:
+            votelst = d["votelst"]
+        if "votekd" in d:
+            votekd = d["votekd"]
+
+        args = {
+            "votekey": votekey,
+            "selkey": selkey,
+            "votefst": votefst,
+            "votelst": votelst,
+            "votekd": votekd,
+        }
+        return args
+
+    def __eq__(self, other):
+        if not isinstance(other, OnlineKeyregTxn):
+            return False
+        return (super(OnlineKeyregTxn, self).__eq__(other) and
+                self.votepk == other.votepk and
+                self.selkey == other.selkey and
+                self.votefst == other.votefst and
+                self.votelst == other.votelst and
+                self.votekd == other.votekd)
+
+
+class OfflineKeyregTxn(Transaction):
+    """
+    Represents an offline key registration transaction.
+    nonpart is implicitly True for this transaction.
+
+    Args:
+        sender (str): address of sender
+        sp (SuggestedParams): suggested params from algod
+        note (bytes, optional): arbitrary optional bytes
+        lease (byte[32], optional): specifies a lease, and no other transaction
+            with the same sender and lease can be confirmed in this
+            transaction's valid rounds
+        rekey_to (str, optional): additionally rekey the sender to this address
+
+    Attributes:
+        sender (str)
+        fee (int)
+        first_valid_round (int)
+        last_valid_round (int)
+        note (bytes)
+        genesis_id (str)
+        genesis_hash (str)
+        group(bytes)
+        type (str)
+        lease (byte[32])
+        rekey_to (str)
+    """
+
+    def __init__(self, sender, sp, note=None, lease=None, rekey_to=None):
+        Transaction.__init__(self, sender, sp, note,
+                             lease, constants.offline_keyreg_txn, rekey_to)
+        if not sp.flat_fee:
+            self.fee = max(self.estimate_size() * self.fee,
+                           constants.min_txn_fee)
+
+    def dictify(self):
+        d = super(OfflineKeyregTxn, self).dictify()
+        od = OrderedDict(sorted(d.items()))
+        return od
+
+    @staticmethod
+    def _undictify(d):
```

Comment:
We could remove this method but I wanted to conform to the style of the other code. My preference is to leave an empty method here.